### PR TITLE
truncatedBackTitle navigationOptions for StackNavigator

### PR DESCRIPTION
**Pull request by Freddy03h**
`truncatedBackTitle` `navigationOptions` for `StackNavigator`

This is to redefined the `'Back'` string default props on the `HeaderBackButton` component. But keeping the truncate behavior on iOS (that is impossible with the `backTitle` option).

It's necessary for translation.

### CODE REVIEW

File: docs/api/navigators/StackNavigator.md [hunk 1]
```diff
@@ -121,6 +121,7 @@ All `navigationOptions` for the `StackNavigator`:
   - `visible` - Boolean toggle of header visibility. Only works when `headerMode` is `screen`.
   - `title` - String or React Element used by the header. Defaults to scene `title`
   - `backTitle` - Title string used by the back button on iOS or `null` to disable label. Defaults to scene `title`
+  - `truncatedBackTitle` - Title string used by the back button on iOS when there is not enough space. Default is `'Back'`
```
Comment (ericvicenti) [inline docs/api/navigators/StackNavigator.md]: "when there is not enough space to display the previous screen's title"

File: src/views/Header.js [hunk 2]
```diff
@@ -149,6 +157,7 @@ class Header extends React.PureComponent<void, HeaderProps, HeaderState> {
       return null;
     }
     const tintColor = this._getHeaderTintColor(props.navigation);
+    const truncatedBackTitle = this._getHeaderTruncatedBackTitle(props.navigation);
```
Comment (grabbou) [inline src/views/Header.js]: `truncatedBackTitle` should be defined on the screen that also defines `backTitle`.

**Comment by grabbou** [comment]
Mostly looking good. Needs re-basing.

File: src/views/Header.js (hunk 2 above)
Comment (Freddy03h) [inline src/views/Header.js]: I'm sorry but I don't understand what should I supposed to do

**Comment by Freddy03h** [comment]
Not easy to rebase, project change a lot since a month and I'm not aware of the changes. I think it will be easier to re-start from a fresh clone.

**Comment by grabbou** [comment]
Yeah. We already got @fson PR into master #1167 👍

**Pull request closed without merging**